sickbeard/tvcache.py

```diff
@@ -171,37 +170,58 @@ def updateCache(self):
                 # set updated
                 self.setLastUpdate()
 
-                cl = []
-                for item in data['entries'] or []:
-                    ci = self._parseItem(item)
-                    if ci is not None:
-                        cl.append(ci)
+                # get last 5 rss cache results
+                recent_results = self.provider.recent_results
+                found_recent_results = 0  # A counter that keeps track of the number of items that have been found in cache
+                stop_at = 3  # Configuration as an error margin, to stop at. The lower the number, the faster it will stop parsing items
 
-                cache_db_con = self._getDB()
+                cl = []
+                index = 0
+                for index, item in enumerate(data['entries'] or []):
+                    if recent_results and item.get('link').strip() in [cached_item['link'].strip() for cached_item in recent_results]:
+                        found_recent_results += 1
+
+                    if found_recent_results >= stop_at:
+                        logger.log(u'Hit the old cached items, not parsing any more for: {0}'.format
+                                   (self.provider_id), logger.ERROR)
+                        break
+                    try:
+                        ci = self._parseItem(item)
+                        if ci is not None:
+                            cl.append(ci)
+                    except UnicodeDecodeError, e:
+                        continue
+
+                cache_db_con = self._get_db()
                 if cl:
                     cache_db_con.mass_action(cl)
 
+                # finished processing, let's save the newest x (index) items and store these in cache with a max of 5 
+                # (overwritable per provider, throug hthe max_recent_items attribute.
+                self.provider.recent_results = data['entries'][0:min(index, self.provider.max_recent_items)]
+
```

Comment:
This shouldn't be done here, but in the provider code, right at the end of the for loop that iterates through the result rows.